src/App.tsx

```diff
@@ -1,26 +1,68 @@
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 import { Loader, TodoFilter, TodoList, TodoModal } from './components';
+import { useEffect, useState, useMemo } from 'react';
+import { useDispatch } from 'react-redux';
+import { AppDispatch } from './app/store';
+import { todosSlice } from './features/todos';
+import { getTodos } from './api';
+import { useAppSelector } from './app/hook';
+import { Todo } from './types/Todo';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+export const App = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const dispatch = useDispatch<AppDispatch>();
+  const todos = useAppSelector(state => state.todosReducer);
+  const currentTodo = useAppSelector(state => state.currentTodoReducer);
+  const filterBy = useAppSelector(state => state.filterReducer.status);
+  const query = useAppSelector(state => state.filterReducer.query);
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+  useEffect(() => {
+    setIsLoading(true);
+    getTodos()
+      .then(todos => dispatch(todosSlice.actions.setTodos(todos)))
```

Comment:
Ensure that the `todosSlice.actions.setTodos` action is correctly defined in your Redux slice. This action should be responsible for updating the state with the fetched todos.